### PR TITLE
feat(unlock-app) - lock detail card & icon selection refactoring

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
This PR adds lock card detail component and a refactor of the modal for customize the NFT.

**Lock card component** 
Screenshot 2022-09-16 at 16 30 32

Screenshot 2022-09-16 at 16 31 01

**Lock card component: loading state** 

Screenshot 2022-09-16 at 16 44 54

**Modal refactoring** 
Screenshot 2022-09-16 at 16 26 17

Screenshot 2022-09-16 at 16 25 33

Screenshot 2022-09-16 at 16 24 59

Screenshot 2022-09-16 at 16 26 00

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by searchableguy** [comment]
Looks a bit too tall. Might wanna make it more squarish in the ratio here.

**Comment by searchableguy** [comment]
I added a comment on adding slight birder and hover effect on the transparent variant of button in an earlier PR review. Can you please do that?

File: unlock-app/src/components/interface/lock/Manage/elements/LockDetailCard.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  ] = useQueries([
+    {
+      queryKey: ['getLock', lockAddress, network],
+      queryFn: getLock,
+    },
+    {
+      queryKey: ['getTokenSymbol', lockAddress, network],
+      queryFn: getTokenSymbol,
+    },
+  ])
```
Comment (searchableguy) [inline unlock-app/src/components/interface/lock/Manage/elements/LockDetailCard.tsx:133]: Good use of useQueries hook for multiple fetch requests 👌

**Review by searchableguy — COMMENTED** [review]
LGTM so far.

Minor nit would be to make the selection button take full space on the card component.

**Comment by kalidiagne** [comment]
> Minor nit would be to make the selection button take full space on the card component.

Screenshot 2022-09-16 at 17 11 13

is done on purpose. after the image selection, the file name will be visible near the button

**Comment by kalidiagne** [comment]
> Looks a bit too tall. Might wanna make it more squarish in the ratio here.

**Before**
Screenshot 2022-09-16 at 17 14 45

**After**
Screenshot 2022-09-16 at 17 14 32

**Comment by searchableguy** [comment]
> > Minor nit would be to make the selection button take full space on the card component.
> 
> Screenshot 2022-09-16 at 17 11 13
> 
> is done on purpose. after the image selection, the file name will be visible near the button 

Yes. I got that. I think it will still be better to show them on different lines or essentially remove the select button when image is selected with a close button.

**Comment by searchableguy** [comment]
I think we can still make it wider. Just roll with it until it is more squarish. 

screen-md is a good size.

**Comment by kalidiagne** [comment]
@searchableguy 

update with suggestions

Screenshot 2022-09-16 at 19 32 18

**Review by searchableguy — APPROVED** [review]

**Pull request merged**